File: Dragon_Dungeons.client/src/models/Campaign.js
import { RepoItem } from "./RepoItem.js"
import { Profile } from "./Account.js"

export class Campaign extends RepoItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.description = data.description
    this.privateNotes = data.privateNotes
    this.publicNotes = data.publicNotes
    this.events = data.events
    this.npcs = data.npcs
    this.monsters = data.monsters
    this.players = data.players
    this.comments = data.comments
    this.initiative = data.initiative
    this.creator = new Profile(data.creator)
    this.creatorId = data.creatorId
  }
}

File: Dragon_Dungeons/Models/Campaign.cs
namespace Dragon_Dungeons.Models;

public class Campaign : RepoItem
{
  public string Name { get; set; }
  public string Description { get; set; }
  public string PrivateNote { get; set; }
  public string PublicNote { get; set; }
  public string Events { get; set; }
  public List<Npc> Npcs { get; set; }
  public string Monsters { get; set; }
  public List<Player> Players { get; set; }
  public string CreatorId { get; set; }
  public Account Creator { get; set; }
}